sim/firesim-lib/src/main/cc/bridges/tracerv.cc

```diff
@@ -162,93 +176,97 @@ void tracerv_t::init() {
     else if (this->trigger_selector == 2)
     {
       write(this->mmio_addrs->triggerSelector, this->trigger_selector);
-      write(this->mmio_addrs->hostTriggerPCStartHigh, this->trace_trigger_start >> 32);
-      write(this->mmio_addrs->hostTriggerPCStartLow, this->trace_trigger_start & ((1ULL << 32) - 1));
-      write(this->mmio_addrs->hostTriggerPCEndHigh, this->trace_trigger_end >> 32);
-      write(this->mmio_addrs->hostTriggerPCEndLow, this->trace_trigger_end & ((1ULL << 32) - 1));
-      printf("TracerV: Collect trace from instruction address %lx to %lx\n", trace_trigger_start, trace_trigger_end);
+      write(this->mmio_addrs->hostTriggerPCStartHigh, this->trigger_start_pc >> 32);
+      write(this->mmio_addrs->hostTriggerPCStartLow, this->trigger_start_pc & ((1ULL << 32) - 1));
+      write(this->mmio_addrs->hostTriggerPCEndHigh, this->trigger_stop_pc >> 32);
+      write(this->mmio_addrs->hostTriggerPCEndLow, this->trigger_stop_pc & ((1ULL << 32) - 1));
+      printf("TracerV: Collect trace from instruction address %lx to %lx\n", trigger_start_pc, trigger_stop_pc);
     }
     else if (this->trigger_selector == 3)
     {
       write(this->mmio_addrs->triggerSelector, this->trigger_selector);
-      write(this->mmio_addrs->hostTriggerStartInst, this->trace_trigger_start & ((1ULL << 32) - 1));
-      write(this->mmio_addrs->hostTriggerStartInstMask, this->trace_trigger_start >> 32);
-      write(this->mmio_addrs->hostTriggerEndInst, this->trace_trigger_end & ((1ULL << 32) - 1));
-      write(this->mmio_addrs->hostTriggerEndInstMask, this->trace_trigger_end >> 32);
+      write(this->mmio_addrs->hostTriggerStartInst, this->trigger_start_insn);
+      write(this->mmio_addrs->hostTriggerStartInstMask, this->trigger_start_insn_mask);
+      write(this->mmio_addrs->hostTriggerEndInst, this->trigger_stop_insn);
+      write(this->mmio_addrs->hostTriggerEndInstMask, this->trigger_stop_insn_mask);
       printf("TracerV: Collect trace with start trigger instruction %x masked with %x, and end trigger instruction %x masked with %x\n",
-              this->trace_trigger_start & ((1ULL << 32) - 1), this->trace_trigger_start >> 32,
-              this->trace_trigger_end & ((1ULL << 32) - 1), this->trace_trigger_end >> 32);
+              this->trigger_start_insn, this->trigger_start_insn_mask,
+              this->trigger_stop_insn, this->trigger_stop_insn_mask);
     }
     else
     {
+      // Biancolin: should we not error here?
```

Comment:
i believe this is to fallback to "no trigger"

https://docs.fires.im/en/latest/Advanced-Usage/Debugging-and-Profiling-on-FPGA/TracerV.html#no-trigger

CC @alonamid

Reply:
Yeah, fallback to "no trigger". I think @zhemao suggested in the past that the default should be "no tracerv" rather than "no trigger", but for now we relay on the enable flag for "no tracerv"

Reply:
I updated the fprintf and made the cycle behavior match that.

Reply:
So ill delete this comment.